UI 컴포넌트 분리 및 코드 정리

JuTaK97 opened the pull request:
이젠 내 코드를 나만 읽지 않는다

JuTaK97 committed:
ModalDrawerWithBottomSheetLayout 분리

JuTaK97 committed:
기존 컴포넌트 패키지 분리

JuTaK97 committed:
바텀시트 개편

JuTaK97 committed:
사용하지 않는 파일 제거

JuTaK97 committed:
BottomNavigation 분리

JuTaK97 committed:
ModalDrawerWithBottomSheetLayout 파라미터 간소화

JuTaK97 committed:
unCheckedNotification을 Flow로 변경

JuTaK97 committed:
TableContext 분리 및 이름 변경

JuTaK97 committed:
HomePage 정리

JuTaK97 committed:
CanvasPalette 분리 및 이름 변경

JuTaK97 committed:
TopBar 획일화

JuTaK97 committed:
스크린샷 관련 함수 분리

JuTaK97 committed:
TimeTable 함수 분리

JuTaK97 committed:
TimetablePage 정리

Hank-Choi commented:
오 `@OptIn` 많이 쓰나보네요
개인적으로 쓰면 뭔가 불안해서 다른식으로 고쳐쓰는데ㅋㅋ

sanggggg commented:
> 오 `@OptIn` 많이 쓰나보네요 개인적으로 쓰면 뭔가 불안해서 다른식으로 고쳐쓰는데ㅋㅋ

그것이 "컴포즈" 니까.
~사실 사이드여서 써보고싶던 api 다 맘대로 가져다씀~

JuTaK97 committed:
BookmarkPage 패키지 변경

JuTaK97 committed:
ChangeThemeBottomSheet 분리

JuTaK97 committed:
CreateTableBottomSheet 분리

JuTaK97 committed:
DrawerTableItem 분리

JuTaK97 committed:
TableMoreActionBottomSheet 분리

JuTaK97 committed:
시간표 이름 변경, 시간표 삭제 다이얼로그 함수화해서 분리

JuTaK97 committed:
ComposableStates 추가

더 나은 이름이 없을까

JuTaK97 committed:
HomeDrawer 패키지 변경 및 정리

JuTaK97 committed:
TimeTablePage 정리

dialog 부분

JuTaK97 committed:
string resource 추가

JuTaK97 committed:
패키지 이름 변경

JuTaK97 committed:
LecturesOfTablePage 정리

JuTaK97 committed:
TableListViewModel 이름 변경

JuTaK97 committed:
LectureDetailViewModel 이름 변경

JuTaK97 committed:
함수 이름에 new 붙은 거 싹다 변경

JuTaK97 committed:
LectureListItem 분리 및 이름 변경

JuTaK97 committed:
SearchTagCell 분리

JuTaK97 committed:
AnimatedLazyRow 분리

JuTaK97 committed:
SearchEditText 분리

JuTaK97 committed:
SearchTopBar에 spacedBy로 horizontalArrangement 지정

JuTaK97 committed:
검색 아이템의 4번째 줄(비고) 아이콘 추가

JuTaK97 committed:
LectureListItem 개편: 강의평 bottomSheet

verifyEmailBeforeApi와 openReviewBottomSheet 함수를 분리해서 코드 가독성과 재사용성을 높였다. (LectureDetailPage와 SearchPage 두 곳에서 사용)

JuTaK97 committed:
Icons에 colorFilter 디폴트 값 변경 및 신규 아이콘 2개 추가

JuTaK97 committed:
string resource 오타 수정

JuTaK97 committed:
이메일 인증 안됐을 때 인증하러 갈지 물어보는 CTA 빼먹은거 추가

JuTaK97 committed:
SearchOptionSheet 불필요한 파라미터 제거

JuTaK97 committed:
관심강좌 제거 다이얼로그 함수 분리

JuTaK97 committed:
강의 겹침 다이얼로그 분리

JuTaK97 committed:
creditStringToLong을 공용 object로 이동

JuTaK97 committed:
DayTimePickerSheet 분리

JuTaK97 committed:
LectureDetailItem 분리

JuTaK97 committed:
SearchPageDialog 작은 변경

- composableState로 묶은 파라미터 사용
- forceAddApi 파라미터 이름 변경

JuTaK97 committed:
LectureState에서 bookmarked 제거

- 각 강의의 bookmarked 여부는 collectAsState하는 bookmarkList로부터 필요할 때 구하기
- LectureDetailPage에서 관심강좌 설정해제를 추가하면서, getBookmarkList api를 막 부르지 말고 bookmarkList로부터 collect하는 형태로 바꾸고 나니 LectureState에서 bookmark가 필요 없어졌다.

JuTaK97 committed:
editingLectureDetail 변수 이름 변경

JuTaK97 committed:
각종 dialog들 분리 (for 가독성)

JuTaK97 committed:
LectureDetailPage를 관리하는 ModeType 추가

JuTaK97 committed:
LectureDetailPage 정리

- ModeType 추가: 과거의 AddMode, ViewMode 이런거를 통합해서 하나의 sealed class인 ModeType으로 관리한다.
- isBookmarked 개선: api를 막 부르는 짓 그만하고 bookmarkList를 collect해서 계산
- 각종 dialog들 별도 함수로 분리해서 가독성 높이기
- 각종 컴포넌트들 최대한 밀도높게 분리해서 코드 줄 수 줄이기
- 비고 editText 개선: 엔터 누르면 focus 해제되지 못하게 하기
- 이와 더불어 EditText들에 KeyboardAction 추가, singleLine 속성 확실히 추가

JuTaK97 committed:
Icon colorFilter 디폴트 개선

JuTaK97 committed:
BookmarkPage 개선

안 쓰는 변수, 불필요한 함수 분리 제거

JuTaK97 committed:
바텀시트 빠르게 여닫을 때 생기는 문제 해결

LaunchedEffect로 bottomSheet가 닫힐 때 content를 비워서 기존 content로 들어있던 composable을 dispose시켰는데, 이렇게 하니 동시성 이슈 때문에 빠르게 여닫으면 새로 열리는 바텀시트의 content가 초기화되는 이슈가 있다.
바텀시트 content를 dispose시키는 경우는 BottomSheet에 LectureDetailPage같이 backHandler가 있는 컴포저블을 넣을 때 뿐이므로, hideAndDispose를 따로 만들어서 사용

JuTaK97 committed:
강의평 탭 WebViewContainer의 compositionLocal 범위를 최소화 및 이름 명확하게 변경

ReviewPage()에만 내려가도록 변경

JuTaK97 committed:
webViewContainer 변수 이름 명확하게 변경 및 불필요한 코드 제거

JuTaK97 committed:
불필요한 CompositionLocal 제거 및 webViewContainer 변수 이름 명확히 변경

JuTaK97 committed:
BottomSheet 소문자로 변경

값을 리턴하는 컴포저블 함수는 소문자로 시작

JuTaK97 committed:
Picker 불필요한 부분 제거

JuTaK97 committed:
CreateTableBottomSheet 개선

- KeyboardAction 추가
- 바텀시트 닫히면 EditText 포커스 해제되도록 설정
- 바텀시트 닫히면 EditText 내용 초기화되도록 변경
- TODO: 바텀시트 닫히면 Picker의 중앙 아이템도 알맞게 바뀌도록 하기

JuTaK97 committed:
bottomSheet 이름 변경 적용

JuTaK97 committed:
DayTimePickerSheet 버그 수정

- 바텀시트를 닫고 다시 열었을 때 요일/시작시간/끝시간이 classTime에 따라recompose되도록 remember에 key 추가
- FIXME: Picker에서 글리치 발생하는 것 임시 수정

JuTaK97 committed:
lint

JuTaK97 committed:
viewmodel 파라미터 디폴트 값로 사용

JuTaK97 committed:
뒤로가기 버그 방지

transition animation 도중에 뒤로가기 동작을 할 경우 backQueue에 있어야 될 parent route의 navEntry가 destroy되는 것으로 추정된다.
우선 backQueue.size로 이를 방지한다.

JuTaK97 committed:
라이브러리 버전 올리기

- 주 목적: androidx.compose.material:material에서 제공하는 ModalBottomSheetLayout에 패치된 버그 수정을 받기 위함 (바텀시트가 붕붕 뜨는 현상 수정)
- 1.4.0-alpha04에 원하는 패치가 있는데, alpha05부터 EditText의 focus 부분에서 크래시 발생
- 아무튼 버전 업을 위해 코틀린 1.8.0으로 변경, 이에 따라 kotlinCompilerExtensionVersion은 1.4.0으로 변경 & Hilt도 2.4.5로 변경
- 하는 김에 Compose 버전도 최신 stable 버전으로 변경 -> 빡센 QA 및 버그 예의주시 필요

Hank-Choi commented:
ㅋㅋㅋㅋㅋ달리네
